src/test/java/io/reactivex/internal/operators/flowable/FlowableGenerateTest.java

```diff
@@ -279,7 +287,71 @@ public void accept(Emitter<Object> e) throws Exception {
                 e.onComplete();
             }
         })
-        .test(1)
-        .assertResult();
+                .test(1)
+                .assertResult();
+    }
+
+    @Test
+    public void flowable_bug_generate() {
```

Comment:
Generate is not meant to be operated from multiple threads so this case is by definition invalid. Besides, multi-threaded races can't be properly detected even with your changes.